syntax = "proto3";
import "profit.proto";
import "profit_factor.proto";
import "expected_payoff.proto";
import "max_drawdown.proto";
import "max_drawdown_rate.proto";
import "relative_drawdown.proto";
import "relative_drawdown_rate.proto";
import "abs_drawdown.proto";
import "abs_drawdown_rate.proto";
import "gross_profit.proto";
import "gross_loss.proto";
import "total_trades.proto";
import "largest_profit_trade.proto";
import "largest_loss_trade.proto";
import "average_profit_trade.proto";
import "average_loss_trade.proto";
import "modeling_quality_percentage.proto";
import "max_consecutive_profit_count.proto";
import "max_consecutive_profit.proto";
import "max_consecutive_loss_count.proto";
import "max_consecutive_loss.proto";
import "max_consecutive_wins_count.proto";
import "max_consecutive_wins_profit.proto";
import "max_consecutive_losses_count.proto";
import "max_consecutive_losses_loss.proto";
import "profit_trades.proto";
import "profit_trades_rate.proto";
import "loss_trades.proto";
import "loss_trades_rate.proto";
import "ave_consecutive_wins.proto";
import "ave_consecutive_losses.proto";
import "short_positions.proto";
import "short_positions_rate.proto";
import "long_positions.proto";
import "long_positions_rate.proto";
import "station_id.proto";

message OptimizationReport{
    Profit profit = 1;
    ProfitFactor profit_factor = 2;
    ExtectedPayoff expected_payoff = 3;
    MaxDrawDown max_drawdown = 4;
    MaxDrawDownRate max_drawdown_rate = 5;
    RelativeDrawDown relative_drawdown = 6;
    RelativeDrawDownRate relative_drawdown_rate = 7;
    AbsDrawDown abs_drawdown = 8;
    AbsDrawDownRate abs_drawdown_rate = 9;
    GrossProfit gross_profit = 10;
    GrossLoss gross_loss = 11;
    TotalTrades total_trades = 12;
    LargestProfitTrade largest_profit_trade = 13;
    LargestLossTrade largest_loss_trade = 14;
    AverageProfitTrade average_profit_trade = 15;
    AverageLossTrade average_loss_trade = 16;
    ModelingQualityPercentage modeling_quality_percentage = 17;
    MaxConsecutiveProfitCount max_consecutive_profit_count = 18;
    MaxConsecutiveProfit max_consecutive_profit = 19;
    MaxConsecutiveLossCount max_consecutive_loss_count = 20;
    MaxConsecutiveLoss max_consecutive_loss = 21;
    MaxConsecutiveWinsCount max_consecutive_wins_count = 22;
    MaxConsecutiveWinsProfit max_consecutive_wins_profit = 23;
    MaxConsecutiveLossesCount max_consecutive_losses_count = 24;
    MaxConsecutiveLossesLoss max_consecutive_losses_loss = 25;
    ProfitTrades profit_trades = 26;
    ProfitTradesRate profit_trades_rate = 27;
    LossTrades loss_trades = 28;
    LossTradesRate loss_trades_rate = 29;
    AveConsecutiveWins ave_consecutive_wins = 30;
    AveConsecutiveLosses ave_consecutive_losses = 31;
    ShortPositions short_positions = 32;
    ShortPositionsRate short_positions_rate = 33;
    LongPositions long_positions = 34;
    LongPositionsRate long_positions_rate = 35;
    StationId station_id = 36;
}
